Compiler bug with in operator and Sum Types

**V version:** V 0.2.4 65d9c8f
**OS:** Ubuntu 20.04

**What did you do?**

Try to use the `in` operator with sumtypes like in this example :
```v
type TokenValue = rune | u64

fn main() {
	val := TokenValue(`+`)
	if val in [TokenValue(`+`), TokenValue(`-`)] {
		println("ok")
	}
}
```

**What did you expect to see?**

The program to run correctly and print "ok"
 
**What did you see instead?**
```
/tmp/v_1000/in_operator_bug.8285416656142212473.tmp.c:11634: error: invalid operand types for binary operation
/tmp/v_1000/in_operator_bug.8285416656142212473.tmp.c:11634: error: invalid operand types for binary operation
...

```

> Note that there is a variant of the bug, don't know if I should create a separated issue. Here is the minimum case :
> 
> ```v
> type TokenValue = rune | u64
> 
> fn main() {
> 	// This is another bug
> 	val := `+`
> 	if val in [TokenValue(`+`), TokenValue(`-`)] {
> 		println("ok")
> 	}
> }
> ```
> 
> And the output :
> 
> ```
> /tmp/v_1000/in_operator_bug.14867944870994764656.tmp.c:11646: error: cannot convert 'unsigned int' to 'struct main__TokenValue'
> ```

I thing something is wrong when it try to look in the array after `in`

if you assign it to an variable it works great

```vlang
type TokenValue = rune | u64

fn main() {
	val := TokenValue(`+`)
	arr := [TokenValue(`+`), TokenValue(`-`)]
	if val in arr {
		println("ok")
		break;
	}
}
```

--- a/vlib/v/gen/c/infix.v
+++ b/vlib/v/gen/c/infix.v
@@ -382,7 +382,8 @@ fn (mut g Gen) infix_expr_in_op(node ast.InfixExpr) {
 	if right.unaliased_sym.kind == .array {
 		if left.sym.kind in [.sum_type, .interface_] {
 			if node.right is ast.ArrayInit {
-				if node.right.exprs.len > 0 {
+				if node.right.exprs.len > 0
+					&& g.table.sym(node.right.expr_types[0]).kind !in [.sum_type, .interface_] {
 					mut infix_exprs := []ast.InfixExpr{}
 					for i in 0 .. node.right.exprs.len {
 						infix_exprs << ast.InfixExpr{

--- a/vlib/v/tests/in_expression_test.v
+++ b/vlib/v/tests/in_expression_test.v
@@ -309,3 +309,10 @@ fn test_in_alias_array() {
 	assert Str('') in [Str(''), Str('a')]
 	assert Struct{} == Struct{}
 }
+
+type TokenValue = rune | u64
+
+fn test_in_array_of_sumtype() {
+	val := TokenValue(`+`)
+	assert val in [TokenValue(`+`), TokenValue(`-`)]
+}

